test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,138 @@
+"use strict";
+
+require('./mockmobile');
+var fs = require('fs');
+var path = require('path');
+var crypto = require('crypto');
+var test = require('tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+
+function deleteFolder(fileLocation, file, cb) {
+  if(file) {
+    fs.unlink(file, function(err) {
+      if(err) {
+        console.error('failed to delete the pkcs12 file - err: ', err);
+      }
+      fs.rmdir(fileLocation);
+      cb();
+    });
+  } else {
+    fs.rmdir(fileLocation);
```

Comment:
Wouldn't it be easier to write:
```javascript
if (file) {
   fs.unlinkSync(file);
}
fs.rmdirSync(fileLocation);
cb();
```

Normally, btw, using sync function is bad but in this case we are running a test framework which is inherently serial anyway, so it's o.k.